src/modules/gifmaker/AnimatedGifRecorder/Recorder.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AnimatedGifRecorder
+{
+    /// <summary>
+    /// Minimized version of ScreenToGif DirectImageCapture
+    /// https://github.com/NickeManarin/ScreenToGif
+    /// </summary>
+    public class Recorder
+    {
+        /// <summary>
+        /// Collection of frame metadata in the current session
+        /// </summary>
+        public List<FrameInfo> Frames { get; private set; } = new List<FrameInfo>();
```

Comment:
is this what we want to do or like a 
get {
   if(frames == null) return new List<FrameInfo>();
    return frames;
}

Reply:
I've seen this pattern quite a lot in the code base, can you tell me why this is preferred over the syntax above?

Reply:
This pattern helps to reduce null-value exceptions by never returning a "null"  Its either whatever value is in there OR it's a new instance.  Depends on the circumstance.